freqtrade/rpc/telegram.py

```diff
@@ -97,340 +19,425 @@ def authorized_only(command_handler: Callable[[Bot, Update], None]) -> Callable[
     :param command_handler: Telegram CommandHandler
     :return: decorated function
     """
-    def wrapper(*args, **kwargs):
+    def wrapper(self, *args, **kwargs):
+        """
+        Decorator logic
+        """
         update = kwargs.get('update') or args[1]
 
         # Reject unauthorized messages
-        chat_id = int(_CONF['telegram']['chat_id'])
+        chat_id = int(self._config['telegram']['chat_id'])
+
         if int(update.message.chat_id) != chat_id:
-            logger.info('Rejected unauthorized message from: %s', update.message.chat_id)
+            self.logger.info(
+                'Rejected unauthorized message from: %s',
+                update.message.chat_id
+            )
             return wrapper
 
-        logger.info('Executing handler: %s for chat_id: %s', command_handler.__name__, chat_id)
+        self.logger.info(
+            'Executing handler: %s for chat_id: %s',
+            command_handler.__name__,
+            chat_id
+        )
         try:
-            return command_handler(*args, **kwargs)
+            return command_handler(self, *args, **kwargs)
         except BaseException:
-            logger.exception('Exception occurred within Telegram module')
-    return wrapper
+            self.logger.exception('Exception occurred within Telegram module')
 
-
-@authorized_only
-def _status(bot: Bot, update: Update) -> None:
-    """
-    Handler for /status.
-    Returns the current TradeThread status
-    :param bot: telegram bot
-    :param update: message update
-    :return: None
-    """
-
-    # Check if additional parameters are passed
-    params = update.message.text.replace('/status', '').split(' ') \
-        if update.message.text else []
-    if 'table' in params:
-        _status_table(bot, update)
-        return
-
-    # Fetch open trade
-    (error, trades) = rpc_trade_status()
-    if error:
-        send_msg(trades, bot=bot)
-    else:
-        for trademsg in trades:
-            send_msg(trademsg, bot=bot)
+    return wrapper
 
 
-@authorized_only
-def _status_table(bot: Bot, update: Update) -> None:
-    """
-    Handler for /status table.
-    Returns the current TradeThread status in table format
-    :param bot: telegram bot
-    :param update: message update
-    :return: None
-    """
-    # Fetch open trade
-    (err, df_statuses) = rpc_status_table()
-    if err:
-        send_msg(df_statuses, bot=bot)
-    else:
-        message = tabulate(df_statuses, headers='keys', tablefmt='simple')
+class Telegram(RPC):
+    """
+    Telegram, this class send messages to Telegram
+    """
+    def __init__(self, freqtrade) -> None:
+        """
+        Init the Telegram call, and init the super class RPC
+        :param freqtrade: Instance of a freqtrade bot
+        :return: None
+        """
+        super().__init__(freqtrade)
+
+        self._updater = None
+        self._config = freqtrade.config
+        self._init()
+
+    def _init(self) -> None:
```

Comment:
Maybe we should rename this method to distinguish better from the constructor. Some names I have in mind are `_bind` and `_start_polling`